### ISSUE TITLE
ansible-galaxy to support collections

### ISSUE THREAD

**Issue by chrismeyersfsu** [ansible/ansible#57081]
##### SUMMARY
`ansible-galaxy` should support collections much like it, currently, supports roles.
* version specification
* `requirements.yml`, like roles, can contain multiple collection + version specifications
* ability to specify the destination directory of the downloaded collection
* Will old-style roles be downloaded and laid down in the collection format?

##### ISSUE TYPE
- Feature Idea

##### COMPONENT NAME
`ansible-galaxy`

##### Needed
* `ansible-galaxy install chrismeyersfsu.ansible_content_example` should function like `mazer install chrismeyersfsu.ansible_content_example`
  * Sources other than galaxy that mazer supports should be supported in `ansible-galaxy` such as a remote git repo or tar url.
* `ansible-galaxy install foobar` when collection `foobar` already exists on disk should be fast i.e. the entire collection should not be needed to be downloaded in order to determine if the version already exists on disk.

##### Nice to Have
* Per-entry results. To know which exact collection/role failed.
* Fast, downloads in parallel.
* Output the result of each individual item easier.
  * an ansible module that is a light wrapper around `ansible-galaxy`
  * --json-output option that outputs the results as json w/ information about each individual collection that was processed (downloaded, already exists, etc).

**Comment by ansibot** [comment]
Files identified in the description:
* [lib/ansible/galaxy](https://github.com/ansible/ansible/blob/devel/lib/ansible/galaxy)
* [lib/ansible/modules/utilities/logic/fail.py](https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/utilities/logic/fail.py)

If these files are inaccurate, please update the `component name` section of the description or use the `!component` bot command.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
cc @dagwieers
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by thaumos** [comment]
Hi @chrismeyersfsu, I'll set some time up with you to chat about this.

**Comment by jctanner** [comment]
ansible-galaxy now has a collections subcommand which fills the requirements outlined in the description.
